medusa/search/proper.py

```diff
@@ -159,6 +159,14 @@ def _get_proper_results(self):  # pylint: disable=too-many-locals, too-many-bran
                 log.info('Skipping non-proper: {name}', {'name': cur_proper.name})
                 continue
 
+            if not cur_proper.show.episodes.get(cur_proper.parse_result.season_number) or \
+                    any([ep for ep in cur_proper.parse_result.episode_numbers
+                         if not cur_proper.show.episodes[cur_proper.parse_result.season_number].get(ep)]):
+                log.info('Skipping proper for wrong season/episode: {name}', {'name': cur_proper.name})
+                if cur_proper.name not in processed_propers_names:
+                    self.processed_propers.append({'name': cur_proper.name, 'date': cur_proper.date})
```

Comment:
yeah, you are right. But somehow this code worked and also didn't worked. Hours later is snatched again. No idea what's wrong